static/usr/lib/systemd/system-generators/rtc-sys-time-init-generator

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+
+# This generator enables rtc-sys-time-init.service by hooking it into
+# systemd-timesyncd.service with symlink in $1/systemd-timesyncd.service.wants 
+# when this is requested by kernel command line flag 'rtc_sys_time_init'.
+
+DEBUG_ENABLE="false"
+SERVICE="rtc-sys-time-init.service"
+SERVICE_PATH="/usr/lib/systemd/system/$SERVICE"
+KERNEL_CMDLINE_ENABLE_FLAG="rtc_sys_time_init"
+HOOK_TO_UNIT="sysinit.target"
+
+log() {
+    echo "$@" >> "/dev/kmsg"
+}
+
+debug() {
+    if [ "$DEBUG_ENABLE" = "true" ]; then
+        log "$@"
+    fi
+}
+
+read_kernel_command_line() {
+    local cmdline="/proc/cmdline"
+    if [ -n "$SYSTEMD_VIRTUALIZATION" ] && [ ! "$SYSTEMD_VIRTUALIZATION" = "none" ]; then
+        debug "Virtualization (vm/container) detected"
+        cmdline="/proc/1/cmdline"
+    fi
+    _RET=$(cat $cmdline)
+    debug "Using command line $cmdline: _RET"
+    return 0
+}
+
+search_command_line() {
+    param=$1
+    if read_kernel_command_line; then
+        cmdline=$_RET
+        for cmd in $cmdline; do
+            if [ "$param" = "$cmd" ]; then
+                return 0
+            fi
+        done
+    fi
+    return 1
+}
+
+main() {
+    local normal_dir="$1" early_dir="$2" late_dir="$3"
+    local link_path="$normal_dir/$HOOK_TO_UNIT.wants"
+    local link_file="$link_path/$SERVICE"
+    local ret=0 
+
+    debug "normal: $normal_dir"
+    debug "early:  $early_dir"
+    debug "late:   $late_dir"
+
+    # Enable rtc-sys-time-init.service if command line contains flag 'rtc-sys-time-init'
+    if search_command_line $KERNEL_CMDLINE_ENABLE_FLAG; then
+        log "Kernel command line requests enable"
+
+        if [ -L "$link_file" ]; then
+            log "Already enabled, no change needed"
+        else 
+            if [ -d "$link_path" ] || mkdir -p "$link_path"; then
+                if ln -snf "$SERVICE_PATH" "$link_file"; then
+                    log "Enabled via $link_file -> $SERVICE_PATH"
+                else
+                    ret=$?
+                    log "Enable failed: cannot create symlink $link_file -> $SERVICE_PATH"
+                fi
+            else
+                ret=$?
+                log "Enable failed: cannot create directory $link_path"
+            fi
+        fi
+    else
+        log "Kernel command line requests disable"
+
+        if [ ! -L "$link_file" ]; then
+            log "Already disabled, no change needed"
+        else
+            if rm -f "$link_file"; then
```

Comment:
I do not think you should remove anything. The issue is that if several generators were to enable for different reason, they should not fight for it. Generators should only enable things. Not disable.

Reply:
Understood, removed link removal.